modal/_utils/function_utils.py

```diff
@@ -198,6 +198,31 @@ def get_globals(self) -> Dict[str, Any]:
         f_globals = {k: func.__globals__[k] for k in f_globals_ref if k in func.__globals__}
         return f_globals
 
+    def class_parameter_format(self) -> "api_pb2.Function.ParameterSerializationFormat.ValueType":
+        if not self.user_cls:
+            return api_pb2.Function.PARAM_SERIALIZATION_FORMAT_UNSPECIFIED
+
+        if os.environ.get("MODAL_STRICT_PARAMETERS"):
```

Comment:
Should we be accessing this via the `config` singleton instead of grabbing the environment variable directly? Also I think that as written this will enter the block when you did

```
MODAL_STRICT_PARAMETERS=0 modal run ...
```

which previously caused some issues — I think the config module has some code to handle it as expected.

Reply:
My thought was that this option is "internal"/"experimental" and not something we want to expose to end users, and will remove before the feature goes live, so it shouldn't matter too much, but yeah it might still be better to put in the config I guess, to prevent foot guns by us internally :)

Reply:
I could be wrong, but I don't think having them in the config makes them all that exposed to end-users; mainly it would show up in modal config show`? But a lot of the other params there have an internal-only flavor already IMO.